geth/api/backend_jail_test.go

```diff
@@ -264,6 +264,9 @@ func (s *BackendTestSuite) TestContractDeployment() {
 		s.FailNow("test timed out")
 	}
 
+	// Wait until callback is fired and `responseValue` is set. Hacky but simple.
```

Comment:
Oh, so was it failing sometimes because of invalid `responseValue`? I have never noticed that...

Reply:
Yes it was. It hasn't failed before just because callbacks were fired synchronously. Now that they're really asynchronous, we somehow need to sync assertions with callbacks invocations.

Reply:
You can refer to this comment to see the whole picture: https://github.com/status-im/status-go/pull/321#discussion_r139318992